freedomofpress/securedrop-workstation
# Initial Qubes 4.2 Updater Integration

## Status
Ready for Review 

(Opened PR due again to the closure of https://github.com/freedomofpress/securedrop-workstation/pull/979 -- see this for background discussion)

TODO:
  - [ ] Consider if it makes sense to update even after a migration was applied (this was not the case before)

Blockers:
  - https://github.com/QubesOS/qubes-issues/issues/8493

## Description of Changes

Fixes #899 . Keeps most of the existing updater logic.

Changes proposed in this pull request:
- updates templates via the CLI Qubes Updater instead of via salt
- removes functions deprecated by the new Qubes updater

## Testing
- [ ] updates failed - no network connection (expectation: updater should quickly bail rather than trying all updates, https://github.com/freedomofpress/securedrop-workstation/issues/633)
- [ ] all updates work
   1. launch updater
   2. Observe templates starting as they are updated
   3. Observe app qubes getting restarted in background
   4. See "successfully completed"
   5. Inspect SDW-launcher logs to ensure correct updating of templates.
- [ ] migration included
   - trigger migration flag and ensure that a migration takes place
- [ ] dom0 updates fail
  - tip: setting sys-firewall's netvm to none helps test this scenario
  - updater should inform of failure. Confirm in logs that dom0 failed to update.
- [ ] vm updates fail
  - tip: use qvm-volume revert on templates to simulate having updates again.
  - tip: disconnect network after 10% (after dom0 updating)
  - updater should inform of failure. Confirm in logs that template failed to update.


## Deployment

Any special considerations for deployment? Will only apply to new installs

## Checklist

### If you have made changes to the provisioning logic


> **Note**: tests don't pass due to the WIP nature of the base 4.2 branch. However, it seems that launcher-specific tests weren't passing already and it was unclear how to run the tests. 

### If you have added or removed files


### If documentation is required
